### PR TITLE
Update the devfile spec to 2.2.1 alpha

### PR DESCRIPTION
**Pull request by schultzp2020**
### What does this PR do?:

1. Updates the devfile spec to 2.2.1 alpha
2. Fixes markdown errors
3. Updates the pr template and issue templates

### Which issue(s) this PR fixes:

N/A

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [ ] Unit/Functional tests
- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 
- [ ] Documentation 
- [ ] Client Impact

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,35 +1,36 @@
-### What does this PR do?:
+# What does this PR do?
```
Comment (amisevsk) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Personal preference: PR descriptions should start at the second level heading (`##`) and use `###` for subheads; the title size is too large and the PR already gets a title set separately.

File: README.md [hunk 2]
```diff
@@ -19,33 +19,35 @@ From these Go sources, several files are generated:
 
 Generated files are created by a build script (see section [How to build](#how-to-build)).
 
-### Devfile 2.0.0 file format
+## Devfile 2.0.0 file format
 
 A Subset of this `DevWorkspace` API defines a structure (workspace template content), which is also at the core of the **Devfile 2.0** format specification.
-For more information about this, please look into the [Devfile support Readme](devfile-support/README.md)
+For more information about this, please look into the [Devfile support README]([devfile-support/README.md](https://github.com/devfile/registry-support/blob/main/README.md))
```
Comment (amisevsk) [inline README.md:25]: This link is broken -- text is `Devfile support README` and link is to `[devfile-support/README.md](https://github.com/devfile/registry-support/blob/main/README.md)`

```suggestion
For more information about this, please look into the [Devfile support README](https://github.com/devfile/registry-support/blob/main/README.md)
```

File: README.md [hunk 3]
```diff
@@ -19,33 +19,35 @@ From these Go sources, several files are generated:
...
 
 ## How to build
 
 In order to build the CRD and the various schemas, you don't need to install any pre-requisite apart from `docker`.
 In the root directory, just run the following command:
 
-```
-./docker-run.sh ./build.sh
+```console
+bash ./docker-run.sh ./build.sh
```
Comment (amisevsk) [inline README.md:41]: Is `bash` necessary here? The scripts have the bash hashbang at the top (`#!/bin/bash`)

**Review by amisevsk — APPROVED** [review]

**Comment by amisevsk** [comment]
Looks like this PR needs `./docker-run.sh ./build.sh` as trimming trailing whitespace has made changes to schema/CRD fields.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (schultzp2020) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Good point, I'll change that.

File: README.md (hunk 3 above)
Comment (schultzp2020) [reply README.md:41]: On non linux systems you can run into issues where you have permission to execute `bash ./*.sh` but not `./*.sh`

Comment (amisevsk) [reply README.md:41]: Makes sense

**Pull request merged**